### PR TITLE
[framework] refresh administrator roles after edit own profile

### PR DESCRIPTION
**Pull request by malyMiso**
| Q             | A
| ------------- | ---
|Description, reason for the PR| refresh administrator roles after edit own profile
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Controller/Admin/AdministratorController.php [hunk 1]
```diff
@@ -125,6 +134,10 @@ public function editAction(Request $request, $id)
             try {
                 $this->administratorFacade->edit($id, $administratorData);
 
+                if ($administrator->getId() === (int)$id) {
```
Comment (grossmannmartin) [inline packages/framework/src/Controller/Admin/AdministratorController.php]: couldn't be typehint in method argument?
```diff
-  public function editAction(Request $request, $id)
+  public function editAction(Request $request, int $id)
```

File: packages/framework/src/Model/Administrator/Security/AdministratorRolesChangedFacade.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public function __construct(TokenStorageInterface $tokenStorage, AdministratorFacade $administratorFacade)
+    {
+        $this->tokenStorage = $tokenStorage;
+        $this->administratorFacade = $administratorFacade;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Administrator\Administrator $administrator
+     */
+    public function refreshAdministratorToke(Administrator $administrator): void
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Administrator/Security/AdministratorRolesChangedFacade.php]:
```diff
-  public function refreshAdministratorToke(Administrator $administrator): void
+  public function refreshAdministratorToken(Administrator $administrator): void
```

not writing the suggestion as usages have to be changed too

File: packages/framework/src/Controller/Admin/AdministratorController.php (hunk 1 above)
Comment (malyMiso) [reply packages/framework/src/Controller/Admin/AdministratorController.php]: fixed

File: packages/framework/src/Model/Administrator/Security/AdministratorRolesChangedFacade.php (hunk 2 above)
Comment (malyMiso) [reply packages/framework/src/Model/Administrator/Security/AdministratorRolesChangedFacade.php]: fixed

**Comment by grossmannmartin** [comment]
Hi, I rebased this pull request to the master branch as now the development version of 9.0.x release is on the master branch.

**Comment by grossmannmartin** [comment]
rebased to current master, added upgrade notes (6cc3ea75b) and removed password from token as it's not present when admin log in (eabb649a5)

**Pull request merged**